packages/wdio-browserstack-service/src/util.ts

```diff
@@ -541,4 +544,12 @@ export function getObservabilityBuildTags(options: BrowserstackConfig & Options.
     return []
 }
 
+export function frameworkSupportsHook(hook: string, framework?: string) {
+    if (framework === 'mocha' && (hook === 'before' || hook === 'after' || hook === 'beforeEach' || hook === 'afterEach')) {
+        return true
+    }
+
+    return false
```

Comment:
if check regarding framework === 'jasmine' not needed?

Reply:
No. For jasmine, we are doing via reporter not through insights_handler. So for jasmine this will not get called.